### ISSUE TITLE
Implement creating new playlists

### ISSUE THREAD

**Issue by abegert** [xou816/spot#413]
Hey,

I am missing the functionality to create a new playlist (I was surprised not to find an already existing issue for this).
I would like to work on this myself. UI wise, I would like to go with what @jannuary had already suggested here: https://github.com/xou816/spot/issues/287#issuecomment-971441309

Anything I should know? My plan so far is basically this:
* Add an item to the sidebar ListBox like 'New Playlist' with a '+' icon
* show some kind of small overlay right next to the 'New Playlist' item with an input box for the name and a button labeled 'Create'
* Write a new method in client.rs that POSTS to the API like this: https://developer.spotify.com/documentation/web-api/reference/#/operations/create-playlist
* call this method in the client when the create button is clicked

Any thoughts?

**Comment by jannuary** [comment]
> show some kind of small overlay right next to the 'New Playlist' item with an input box for the name and a button labeled 'Create'

I'm not too sure what you mean by overlay - there are popovers and there are dialogs.

**Comment by abegert** [comment]
> > show some kind of small overlay right next to the 'New Playlist' item with an input box for the name and a button labeled 'Create'
> 
> I'm not too sure what you mean by 'overlay' - there are popovers and there are dialogs.

Sorry, I was just trying to describe it in my own words and I'm not familiar with all the GTK lingo yet.
Popover looks like what I was thinking.

**Comment by jannuary** [comment]
> Sorry, I was just trying to describe it in my own words and I'm not familiar with all the GTK lingo yet. Popover looks like what I was thinking.

Popovers don't usually have "Confirm" and such buttons :p
I'm thinking of something more of the lines that you create the playlist → you get sent to it → edit mode (outlined in #354) turns on, with the focus on the title entry. That way, you aren't explicitly required to name your playlist, but you can still do so quickly.

There also should be a "New Playlist" entry in the selection mode, and in that case, an unnamed playlist is transparently created, but not opened.

**Comment by abegert** [comment]
> > Sorry, I was just trying to describe it in my own words and I'm not familiar with all the GTK lingo yet. Popover looks like what I was thinking.
> 
> Popovers don't usually have "Confirm" and such buttons :p I'm thinking of something more of the lines that you create the playlist → you get sent to it → edit mode (outlined in #354) turns on, with the focus on the title entry. That way, you aren't explicitly required to name your playlist, but you can still do so quickly.
> 
> There also should be a "New Playlist" entry in the selection mode, and in that case, an unnamed playlist is transparently created, but not opened.

Also a good idea, that is actually how the official desktop client does it, too. Sounds good to me!

Just to make sure I understand: In selection mode, you would select one or more songs, click 'New Playlist', which will then create a new playlist with the selected songs added to it. Maybe it makes sense to then show a popover that explains that a new playlist has been created, including it's name, such as 'New Playlist \#42'? The view would stay the same so you can keep adding titles.

**Comment by jannuary** [comment]
> Maybe it makes sense to then show a popover that explains that a new playlist has been created, including it's name, such as 'New Playlist #42'? 

Right, haven't thought about this 😅 Wouldn't be a popover though, but a [Toast](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.ToastOverlay.html) with an "Open" button.

**Comment by xou816** [comment]
Hi! Thanks for hashing this out together already :) 

On the technical side, I think you got it!

On the design side, may I suggest something different? I think a New playlist item in the sidebar is fine indeed, with a popover similar to the rename popover in Nautilus for instance. Sending the user directly to it might not be interesting since the primary way to add content to the playlist is to add songs from some other place. This also allows us to handle the edit mode of playlists later and have less work for an initial implementation of playlist creation :p

**Comment by jannuary** [comment]
Hm, I guess if Nautilus is doing it, this might be fine
Although I do wonder if having a popover like this in a sidebar is a little awkward

**Comment by xou816** [comment]
![image](https://user-images.githubusercontent.com/6277494/147590237-a9001e4d-d203-48fb-8ca1-72f1114999fe.png)

^ to clarify what I meant

**Comment by xou816** [comment]
![image](https://user-images.githubusercontent.com/6277494/147590311-9f339b25-aae5-466b-abc8-3768acc1bc10.png)

^ to clarify what I meant

**Comment by jannuary** [comment]
> ^ to clarify what I meant

Ah, I see, fine by me

**Comment by abegert** [comment]
Looks good to me! I haven't had the time to start working on this and I'm happy that we were able to clarify it.
